source/assets/js/components/fef-swipeable-area.js

```diff
@@ -27,166 +24,250 @@ export function init(interactionMeasureString = '') {
  */
 export class FefSwipeableArea {
 
-    constructor($element, interactionMeasureString = '') {
+    constructor($element) {
+        // DOM references cached here for performance and readability reasons
         this.$element = $element;
-        this.$innerContainer = $(`.${INNER_CONTAINER_CLASS}`, this.$element);
-        this.$items = [];
-        this.itemPositions = [];
+        this.$scrollContainer = $(`.${OUTER_CONTAINER_CLASS}`, this.$element);  // scrollable parent of the <ul>
+        this.$innerContainer = $(`.${INNER_CONTAINER_CLASS}`, this.$element);   // <ul>, the direct parent of the teasers
+        this.$items = $(`.${ITEM_CLASS}`, this.$innerContainer);                // <li>, the wrappers around the teaser elements
         this.$buttonBack = null;
         this.$buttonForward = null;
-        this.interactionMeasureString = interactionMeasureString;
 
-        this.visibleClass = null;
-        this.hiddenClass = null;
+        this.itemPositions = []; // save the items' positions so we don't have to collect them on every event we need them. Will be updated on resize and when the items change
+        this.containerDimensions = {};
 
-        this.oldScrollLeft = this.$innerContainer.scrollLeft();
-        this.isPageBackClick = false;
-        this.isPageForwardClick = false;
+        this.isTouchSupported = FefTouchDetection.isTouchSupported();
+        this.isMobile = this.checkIfIsMobile();
 
-        this.initOnce();
-        this.init();
+        this.useHinting = !!this.$element.data('swipeable-hinting'); // hinting can be enabled by using "data-swipeable-hinting" on the root component
+
+        // let's go!
+        this.initialSetup();
     }
 
-    initOnce() {
-        this.initItemCheck();
+    /**
+     * Determine if this is a mobile breakpoint; "mobile" meaning a screenwidth
+     * less than 1024px and without touch support.
+     * 
+     * @returns {boolean}
+     */
+    checkIfIsMobile() {
+        return $(window).outerWidth() < DESKTOP_WIDTH && !this.isTouchSupported;
```

Comment:
Verstehe ich nicht. mobile device ohne Touch?

Reply:
> "mobile" meaning a screenwidth less than 1024px and without touch support.

meinst du das naming ist kacke? Konnte keinen besseren Namen finden :(

Reply:
Ne, wieso muss beim Mobile device das Touch nicht supported sein?

Reply:
Hm, da muss ich nochmal nachschauen.
Meine Auffassung war, dass es um das Bedienen geht. Mit einem Finger kann man ja bequem swipen, wenn touch supported wird. Das _sollte_ ja gemäss unseren Auffassungen von Breakpoints ja auf Mobile- und Tablet-Breakpoint der Fall sein.

Das haben wir hier eingefügt: https://github.com/mmz-srf/srf-frontend-framework/commit/fd0a3750db3b05568cb5e5aa120f52ebe38b8795#diff-07067b55c1fdc64e7df105a599bd8d5bR57

Im Ticket (https://srfmmz.atlassian.net/browse/SRFCMSAL-2038) ist von Flackern und Performanz die Rede.

Aber ich gehe dem noch nach 👍

Reply:
Hab jetzt mal bei Lutz nachgefragt, aber es ist immer noch die gleiche Logik wie früher. Wenn eine Änderung kommt, würde ich sie entweder hier oder in einem neuen PR tun.

Wahrscheinlich dann sowas:

```
shouldShowButtons() {
    return FefResponsiveHelper.isDesktopUp() && !this.isTouchSupported;
}
```

--> Buttons nie auf Mobile und auf Desktop nur wenn kein Touch-Support

Reply:
Lutz findet's so gut, und du?

Reply:
Ja, jetzt, wo die methode auch das beschreibt was sie macht "shouldShowButtons" ist es irgendwie klar